lib/workers/repository/model/commit-message.ts

```diff
@@ -21,34 +27,54 @@ export class CommitMessage {
     return `${prefix}${CommitMessage.SEPARATOR}`;
   }
 
-  public setMessage(message: string): void {
-    this.message = (message || '').trim();
+  toString(): string {
+    const parts: ReadonlyArray<string | undefined> = [
+      this.title,
+      this.body,
+      this.footer,
+    ];
+
+    return parts.filter(Boolean).join('\n\n');
   }
 
-  public setCustomPrefix(prefix?: string): void {
-    this.prefix = (prefix ?? '').trim();
+  get title(): string {
+    return [CommitMessage.formatPrefix(this.prefix), this.formatSubject()]
+      .join(' ')
+      .trim();
   }
 
-  public setSemanticPrefix(type?: string, scope?: string): void {
-    this.prefix = (type ?? '').trim();
+  toJSON(): CommitMessageJSON {
+    return {
+      body: this.body,
+      footer: this.footer,
+      subject: this.subject,
+    };
+  }
 
-    if (scope?.trim()) {
-      this.prefix += `(${scope.trim()})`;
-    }
+  setBody(body?: string): void {
```

Comment:
I'm also not happy about setter and getter functions. This is more a java common that a JavaScript. I still prefer JavaScript properties instead.

@rarkins @JamieMagee WDYT?

Reply:
Yeah, not a fan of this style either. TypeScript has `get` and `set` keywords for this purpose.

```ts
#_body: string;

set body(body: string) {
	this.#_body = body.trim();
}
```

Reply:
Refactored to getters and setters